gateway/api/authentication.py

```diff
@@ -28,81 +26,24 @@ class CustomToken:
-
+    def authenticate(self, request):
         quantum_user = None
-        token = None
-        if auth_header is not None and auth_url is not None:
-            token = auth_header.split(" ")[-1]
-
-            auth_data = safe_request(
-                request=lambda: requests.post(
-                    auth_url,
-                    json={settings.SETTINGS_TOKEN_AUTH_TOKEN_FIELD: token},
-                    timeout=60,
-                )
-            )
-            if auth_data is not None:
-                user_id = auth_data.get(settings.SETTINGS_TOKEN_AUTH_USER_FIELD)
-
-                verification_data = safe_request(
-                    request=lambda: requests.get(
-                        verification_url,
-                        headers={"Authorization": auth_data.get("id")},
-                        timeout=60,
-                    )
-                )
-
-                if verification_data is not None:
-                    verifications = []
-                    for (
-                        verification_field
-                    ) in settings.SETTINGS_TOKEN_AUTH_VERIFICATION_FIELD.split(";"):
-                        nested_field_value = verification_data
-                        for nested_field in verification_field.split(","):
-                            nested_field_value = nested_field_value.get(nested_field)
-                        verifications.append(nested_field_value)
-
-                    verified = all(verifications)
-
-                    if user_id is not None and verified:
-                        quantum_user, created = QuantumUserProxy.objects.get_or_create(
-                            username=user_id
-                        )
-                        if created:
-                            logger.info("New user created")
-                        quantum_user.update_groups(auth_data.get("id"))
-
-                    elif user_id is None:
-                        logger.warning("Problems authenticating: No user id.")
-
-                    else:  # not verified
-                        logger.warning("Problems authenticating: User is not verified.")
-
-                else:  # verification_data is None
-                    logger.warning(
-                        "Problems authenticating: No verification data returned from request."
-                    )
-
-            else:  # auth_data is None
-                logger.warning(
-                    "Problems authenticating: No authorization data returned from auth url."
-                )
-
-        elif auth_header is None:
+        authorization_token = None
+        auth_header = request.META.get("HTTP_AUTHORIZATION")
+        if auth_header is None:
             logger.warning(
-                "Problems authenticating: User did not provide authorization token."
+                "Problems authenticating: user did not provide authorization token."
             )
+            return quantum_user, CustomToken(authorization_token.encode())
```

Comment:
If `authorisation_token` is `None` at this point, wouldn't `authorisation_token.encode()` raise an error?

Reply:
Good catch @paaragon , thank you!

Reply:
Solved in [d779f7e](https://github.com/Qiskit/qiskit-serverless/pull/1586/commits/d779f7e6d7ddba235211e9602060d0bf2666acc0)